src/runtime/server/middleware/session/index.ts

```diff
@@ -113,14 +122,24 @@ const getSession = async (event: H3Event): Promise<null | Session> => {
   return session
 }
 
+const touchSession = (session: Session, event: H3Event) => {
+  const now = new Date()
+  session.createdAt = now
+  safeSetCookie(event, SESSION_COOKIE_NAME, session.id, now)
```

Comment:
Same here: please avoid modifying input arguments of a function and produce side effects. We're aiming for our objects to be immutable and our functions to produce 0 side effects. 

Instead, you can create and return a new session object from this function
```suggestion
  const now = new Date()
  safeSetCookie(event, SESSION_COOKIE_NAME, session.id, now)
  return { ...session, createdAt: now }
```

Nit: I would then call this function `updateSessionExpirationDate` and ideally would move it to a separate file, since this logic is only about optional configuration, therefore doesn't belong to the main index file

Reply:
Moving `updateSessionExpirationDate` to a separate file needs `safeSetCookie` to be accessible from outside of the index file, if that is desired?

Reply:
> Moving updateSessionExpirationDate to a separate file needs safeSetCookie to be accessible from outside of the index file, if that is desired?

Well, you can also provide `safeSetCookie` via a functional argument (while this not being a good pattern I would recommend). Do you see downsides on moving `safeSetCookie` accessible outside of the index file or just because of additional refactoring this requires?

Reply:
I don't see the benefit for moving logic of modifying the session outside of the session index file, so I would keep `updateSessionExpirationDate` there.